cmd/kubeadm/app/cmd/upgrade/apply.go

```diff
@@ -108,34 +137,18 @@ func NewCmdApply(apf *applyPlanFlags) *cobra.Command {
 //   - Uploads the newly used configuration to the cluster ConfigMap
 func runApply(flags *applyFlags, userVersion string) error {
 
-	// Start with the basics, verify that the cluster is healthy and get the configuration from the cluster (using the ConfigMap)
-	klog.V(1).Infoln("[upgrade/apply] verifying health of cluster")
-	klog.V(1).Infoln("[upgrade/apply] retrieving configuration from cluster")
-	client, versionGetter, cfg, err := enforceRequirements(flags.applyPlanFlags, flags.dryRun, userVersion)
```

Comment:
this function `enforceRequirements` was an issue because it not only got the `client`, `versionGetter` and `cfg` but it also ran preflight checks and stuff and needed split out to enable the building of the data structure and running preflight checks.